### PR TITLE
Host pagination debug

### PR DESCRIPTION
**Pull request by supunvindula**

### CODE REVIEW

File: src/pages/Hosts/index.js [hunk 1]
```diff
@@ -16,22 +16,21 @@ const PAGE_SIZE = 10;
 export default function Hosts() {
   const history = useHistory();
   const evernode = useEvernode();
+  const {nextPageToken, onChangeNextPageToken, pageQueue, onChangePageQueue} = useEvernode();
```
Comment (chalith) [inline src/pages/Hosts/index.js:19]: Shall we rename `onChangeNextPageToken` to `updateNextPageToken` and `onChangePageQueue` to `updatePageToken`

File: src/pages/Hosts/index.js [hunk 2]
```diff
@@ -103,15 +107,17 @@ export default function Hosts() {
 
   const handleNextClick = useCallback(() => {
     setIsHostsLoading(true);
-    setPageQueue([...pageQueue, nextPageToken]);
+    onChangePageQueue(nextPageToken);
```
Comment (chalith) [inline src/pages/Hosts/index.js:110]: updatePageQueue([...pageQueue, nextPageToken]);

File: src/pages/Hosts/index.js [hunk 3]
```diff
@@ -103,15 +107,17 @@ export default function Hosts() {
...
     loadHosts(nextPageToken);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [loadHosts, pageQueue, nextPageToken]);
 
   const handlePrevClick = useCallback(() => {
     setIsHostsLoading(true);
     const prevPageToken = pageQueue.length > 1 ? pageQueue[pageQueue.length - 2] : null;
-    setPageQueue(pageQueue.slice(0, pageQueue.length - 1));
+    onChangePageQueue(pageQueue.slice(0, pageQueue.length - 1));
```
Comment (chalith) [inline src/pages/Hosts/index.js:118]: updatePageQueue(pageQueue.slice(0, pageQueue.length - 1));

File: src/services/Evernode.js [hunk 4]
```diff
@@ -11,6 +11,27 @@ const { createContext, useContext } = React;
...
 export const EvernodeProvider = (props) => {
     const [loading, setLoading] = useState(true);
+    const [nextPageToken, setNextPageToken] = useState(null);
+    const [pageQueue, setPageQueue] = useState([]);
+
+    const onChangeNextPageToken = (token) => {
+        setNextPageToken(token);
+    }
+
+    const onChangePageQueue = (pageToken, isRemove = false) => {
```
Comment (chalith) [inline src/services/Evernode.js:21]:
```
const updatePageQueue = (updatedPageQueue) => {
    setPageQueue(updatedPageQueue);
}
```

**Review by chalith — APPROVED** [review]

**Review by kithminisg — APPROVED** [review]

**Review by kasunkavinda — APPROVED** [review]

**Pull request merged**